app/lib/shared/cvss2Tools.js

```diff
@@ -0,0 +1,145 @@
+import cvss2js from 'cvss2js'
+import * as cvss2 from '../../../csaf-validator-lib/lib/shared/cvss2.js'
+
+/** @typedef {(typeof cvss2.mapping)[number][0]} CVSSField */
+/** @typedef {(typeof cvss2.mapping)[number][1]} CVSSVectorStringKey */
+/** @typedef {keyof (typeof cvss2.mapping)[number][2]} CVSSValue */
+
+/**
+ * @param {Vector} vector
+ * @template {Record<string, string>} Vector
+ */
+export function vectorUpdateFromVectorString(vector) {
+  const { vectorString } = vector
+  if (typeof vectorString !== 'string') return vector
+  const vectorStringMap = vectorString
+    .split('/')
+    .map((k) => {
+      const [key, value] = k.split(':')
+      const mapping = cvss2.mapping.find((m) => m[1] === key)
+      if (!mapping) return null
+      const valueMapping = new Map(
+        Array.from(
+          /** @type {Iterable<[String, { id: string; score: number }]>} */ (
+            Object.entries(mapping[2])
+          )
+        ).map(([key, value]) => [value.id, key])
+      )
+
+      const vectorKeyName = mapping[0]
+      const vectorField = valueMapping.get(value)
+      if (vectorField === undefined) return null
+      return /** @type {const} */ ([vectorKeyName, vectorField])
+    })
+    .filter(
+      /**
+       * @returns {entry is [readonly CVSSField, string]}
+       */
+      (entry) => Boolean(entry)
+    )
+
+  // Apply updates only if necessary to keep referential transparency
+  return vectorStringMap.reduce(
+    (vector, entry) =>
+      vector[entry[0]] !== entry[1]
+        ? { ...vector, [entry[0]]: entry[1] }
+        : vector,
+    vector
+  )
+}
+
+/**
+ * @param {Record<string, string | undefined>} vector
+ * @returns {string}
+ */
+export function vectorGetVectorString(vector) {
+  const vectorMapping = new Map(Object.entries(vector))
+  const f = cvss2.mapping
```

Comment:
I'd suggest to use a slightly more descriptive name than `f`. Maybe sth. like `vectorFields` would work.